spec/lib/cyclonedx/report_go_dep_cyclonedx_spec.rb

```diff
@@ -0,0 +1,172 @@
+require_relative '../../spec_helper'
+require 'json'
+
+describe Cyclonedx::ReportGoDep do
+  describe "#run" do
+    it 'should report all the deps using go.sum if available' do
+      repo = Salus::Repo.new('spec/fixtures/report_go_sum')
+      scanner = Salus::Scanners::ReportGoDep.new(repository: repo, config: {})
+      scanner.run
+
+      golang_cyclonedx = Cyclonedx::ReportGoDep.new(scanner.report)
+      expect(golang_cyclonedx.build_components_object[0..2]).to match_array(
+        [
+          {
+            "bom-ref": "pkg:golang/github.cbhq.net/c3/bls12-381",
+            "type": "library",
+            "group": "",
+            "name": "github.cbhq.net/c3/bls12-381",
+            "version": "v0.0.0-20210114210818-577bfdc5cb9c",
+            "purl": "pkg:golang/github.cbhq.net/c3/bls12-381",
+            "properties": [
+              {
+                "key": "source",
+                "value": "N/A for go.mod/go.sum dependencies"
+              },
+              {
+                "key": "dependency_file",
+                "value": "go.sum"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:golang/github.cbhq.net/c3/bls12-381",
+            "type": "library",
+            "group": "",
+            "name": "github.cbhq.net/c3/bls12-381",
+            "version": "v0.0.0-20210114210818-577bfdc5cb9c",
+            "purl": "pkg:golang/github.cbhq.net/c3/bls12-381",
+            "properties": [
+              {
+                "key": "source",
+                "value": "N/A for go.mod/go.sum dependencies"
+              },
+              {
+                "key": "dependency_file",
+                "value": "go.sum"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:golang/github.com/davecgh/go-spew",
+            "type": "library",
+            "group": "",
+            "name": "github.com/davecgh/go-spew",
+            "version": "v1.1.0",
+            "purl": "pkg:golang/github.com/davecgh/go-spew",
+            "properties": [
+              {
+                "key": "source",
+                "value": "N/A for go.mod/go.sum dependencies"
+              },
+              {
+                "key": "dependency_file",
+                "value": "go.sum"
+              }
+            ]
+          }
+        ]
+      )
+    end
+
+    it 'should report all the deps in the Gopkg.lock if present' do
+      repo = Salus::Repo.new('spec/fixtures/report_go_dep')
+      scanner = Salus::Scanners::ReportGoDep.new(repository: repo, config: {})
+      scanner.run
+
+      golang_cyclonedx = Cyclonedx::ReportGoDep.new(scanner.report)
+      expect(golang_cyclonedx.build_components_object).to match_array(
+        [
+          {
+            "bom-ref": "pkg:golang/github.com/PagerDuty/go-pagerduty",
+            "type": "library",
+            "group": "",
+            "name": "github.com/PagerDuty/go-pagerduty",
+            "version": nil,
```

Comment:
I think setting nil may result in an invalid cyclonedx report, can you add in a test to `cyclonedx_report_spec` similar to https://github.com/coinbase/salus/blob/master/spec/lib/cyclonedx/cyclonedx_report_spec.rb#L13 to test if this will raise an error.

Reply:
will do!